### PR TITLE
(1) update MOM6 to 20230206 main branch (2) remove cpld_esmfthread_P8 (3) hafs bug fix

### LINKED ISSUE #1588: hafs_regional_storm_following_1nest_atm_ocn failed with debug test
## Description
Hafs group reported that the hafs_regional_storm_following_1nest_atm_ocn test failed in debug mode. The error showed in the moving nest lat info in the module_wrt_grid_comp.F90. 

### To Reproduce:
From Biju:
I ran regression tests with "hafs_regional_storm_following_1nest_atm_ocn_wav" and "hafs_regional_storm_following_1nest_atm_ocn" with DEBUG flag on.

The first one (hafs_regional_storm_following_1nest_atm_ocn_wav) completed successfully, but the second one(hafs_regional_storm_following_1nest_atm_ocn) failed with the error: 

forrtl: error (72): floating overflow
Image              PC                Routine            Line        Source
fv3.exe            000000000FDCDAAB  Unknown               Unknown  Unknown
libpthread-2.31.s  000014EB7A0AF8C0  Unknown               Unknown  Unknown
fv3.exe            0000000005C1454B  module_wrt_grid_c         759  module_wrt_grid_comp.F90

One difference I noticed between these two runs is that in the first one, the output_grid is (rotated_latlon,rotated_latlon), while in the second one, the output_grid is (regional_latlon, regional_latlon_moving).

### LINKED ISSUE #1598: drop cpld_esmfthreads_p8 test
## Description

The ``cpld_esmfthreads_p8`` test can be dropped after PR #1523 which made all tests use esmf-managed threading. 

## Solution

Remove the test from rt.conf and tests/tests.

### LINKED ISSUE #1601: update MOM6 to its main branch 20230206 updating
MOM6 main branch is updated on 20230206, this is based on EMC ungridded dimension pstokes PR (https://github.com/mom-ocean/MOM6/pull/1591), but being added through cherry-pick the ci failure fixing from dev/gfdl.
need to make corresponding updating in dev/emc and ufs-weather-model

### PR DESCRIPTION
**Pull request by jiandewang**
## Description
MOM6 main branch is updated on 20230206, this is based on EMC ungridded dimension pstokes PR (https://github.com/mom-ocean/MOM6/pull/1591), but being added through cherry-pick the ci failure fixing from dev/gfdl.
need to make corresponding updating in dev/emc and ufs-weather-model. No answer change is expected.

### Input data additions/changes
- [x] No changes are expected to input data.
- [ ] There will be new input data. 
- [ ] Input data will be updated. 

### Anticipated changes to regression tests:
- [ ] No changes are expected to any regression test. 
- [x] Changes are expected to the following tests: 
a new test is added, hafs_regional_storm_following_1nest_atm_ocn_debug

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [x] FV3
- [ ] GOCART
- [ ] HYCOM
- [x] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Combined with PR's (If Applicable):
- #1592 

## Commit Queue Checklist:

- [x] Link PR's from all sub-components involved
- [ ] Confirm reviews completed in sub-component PR's
- [ ] Add all appropriate labels to this PR.
- [x] Run full RT suite on either Hera/Cheyenne with both Intel/GNU compilers
- [ ] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Linked PR's and Issues:

- depends on https://github.com/NOAA-EMC/fv3atm/pull/622
- depends on https://github.com/NOAA-EMC/MOM6/pull/111
- fixes #1598 
- fixes #1588
- fixes #1601

## Testing Day Checklist:

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - Intel
    - [x] Hera
    - [x] Orion
    - [x] Jet
    - [x] Gaea
    - [x] Cheyenne
  - GNU
    - [x] Hera
    - [x] Cheyenne
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment
[HERA-gun-log.pdf](https://github.com/ufs-community/ufs-weather-model/files/10700914/HERA-gun-log.pdf)

### CODE REVIEW

**Comment by jiandewang** [comment]
[HERA_intel-log.pdf](https://github.com/ufs-community/ufs-weather-model/files/10700922/HERA_intel-log.pdf)

**Comment by DeniseWorthen** [comment]
Do we need a new baseline? Can't we just add the one new debug test to the existing baseline?

**Comment by jkbk2004** [comment]
> Do we need a new baseline? Can't we just add the one new debug test to the existing baseline?

Sure, I can revert the BL_DATE change and do manually add hafs_regional_storm_following_1nest_atm_ocn_debug.

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA I am going to manually create hafs_regional_storm_following_1nest_atm_ocn_debug and copy over to develop-20230201

**Comment by support-epic** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: RT
[RT] Repo location: /glade/scratch/epicufsrt/autort/jenkins/autort/pr/1235626959/20230209161513/ufs-weather-model
Please make changes and add the following label back: cheyenne-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: RT
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/1235626959/20230209224515/ufs-weather-model
[RT] Error: Test cpld_control_p8 046 failed in run_test failed 
[RT] Error: Test cpld_debug_p8 048 failed in run_test failed 
Please make changes and add the following label back: hera-gnu-RT

**Comment by jkbk2004** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/epic-ps/jongkim/autort/pr/1235626959/20230209203013/ufs-weather-model
Please make changes and add the following label back: orion-intel-RT

**Comment by jkbk2004** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/epic-ps/jongkim/autort/pr/1235626959/20230209204515/ufs-weather-model
Please make changes and add the following label back: orion-intel-RT

**Comment by DeniseWorthen** [comment]
@jkbk2004 @BrianCurtis-NOAA RTs were started prior to obtaining PR approval for dependent MOM6 PR.

**Comment by jkbk2004** [comment]
Once test evidences and component/app level approvals are available, we should let RT run. HAFS team approved #1592. In this PR, this condition is met. If we are too restrictive to sequential process, same result that whole process would slow down.

**Comment by jkbk2004** [comment]
Plus, a minimal scope of code change on mom6 side and my manual initial test ran ok.

**Comment by jkbk2004** [comment]
I think setting ORT or Jenkins-CI or single RT test as requirement would make a sense before the RT commit. We can request to either PR owner or one of EPIC CMs.

**Comment by BrianCurtis-NOAA** [comment]
@jkbk2004 The idea behind waiting for completed approvals (all requested reviewers) in sub components is so that we do not risk running tests more than once due to a reviewer requesting changes to part of the code. A good review will take time, as they should, so that part cannot be rushed or skipped. The resources we have are not unlimited and also shared, so it makes sense to hold off until all reviews are completed.

What I disagree with is that there's anything wrong with the process slowing down. Once all requirements are met for the commit queue, then the process from that to merge is going to be faster than it was before.

**Comment by DeniseWorthen** [comment]
I agree that for this PR, the changes on MOM6 are trivial. However, we did institute a procedure which says that component level PRs will be approved prior to starting RT testing. Or am I misunderstanding the PR template checklist?

**Comment by jkbk2004** [comment]
I think we can agree on the condition of commit first: 1) an evidence of test and 2) approval of code change from component/app CM (or SME). Of course, even this condition met, there is still chance that additional issues can be found during full RT tests.

**Comment by junwang-noaa** [comment]
@jkbk2004 I do suggest finishing the commit queue checklist shown in the PR template. It helps to maintain a consistent and transparent commit process that avoids lots of confusions and issues, it also helps streamline the PR commits and eventually speed up the commits. Thanks

**Comment by jkbk2004** [comment]
> @jkbk2004 I do suggest finishing the commit queue checklist shown in the PR template. It helps to maintain a consistent and transparent commit process that avoids lots of confusions and issues, it also helps streamline the PR commits and eventually speed up the commits. Thanks

Sure!

**Comment by BrianCurtis-NOAA** [comment]
Also it's preferable that a full review is completed on the UFSWM PR prior to testing. Even if its as simple as reviewing the .gitmodules and make sure hash's make sense.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by zach1221** [comment]
Please see jenkins-ci logs attached.
[ufs-weather-model » ort-docker-pipeline » PR-1606 #1 Console [Jenkins].pdf](https://github.com/ufs-community/ufs-weather-model/files/10709102/ufs-weather-model.ort-docker-pipeline.PR-1606.1.Console.Jenkins.pdf)

**Comment by jkbk2004** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/epic-ps/jongkim/autort/pr/1235626959/20230211081515/ufs-weather-model
Please make changes and add the following label back: orion-intel-RT

**Comment by jkbk2004** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/epic-ps/jongkim/autort/pr/1235626959/20230211091514/ufs-weather-model
Please make changes and add the following label back: orion-intel-RT

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA @SadeghTabas-NOAA We can start merging process after wcoss2 tests.

**Comment by BrianCurtis-NOAA** [comment]
The new test needs to be disabled on Acorn. The new test calls for 640 nodes on a system that only has 32. This new test seems to need a very large amount of resources for a regression test especially when it only runs for 1 hour.

**Comment by junwang-noaa** [comment]
@BrianCurtis-NOAA Please feel free to turn it off on acorn.   @BinLiu-NOAA FYI.

**Comment by BrianCurtis-NOAA** [comment]
> The new test needs to be disabled on Acorn. The new test calls for 640 nodes on a system that only has 32. This new test seems to need a very large amount of resources for a regression test especially when it only runs for 1 hour.

Sure I will turn it off in my commit for the Acorn log coming in a little bit.

**Comment by BrianCurtis-NOAA** [comment]
Nevermind about the Acorn issue. It seems Acorn itself it having an issue. I read the wiki wrong.

**Comment by BrianCurtis-NOAA** [comment]
Please skip Acorn for this PR. The builds go OK, but the runs don't start. I'm looking into it. I am leaning towards a temporary machine issue or limit.

File: tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug
Comment (DeniseWorthen) [inline tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: The control directory should be the debug test.

Comment (junwang-noaa) [reply tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: Oops, you are right! Sorry, I missed that. @jiandewang would you please change this to hafs_regional_storm_following_1nest_atm_ocn_debug

Comment (DeniseWorthen) [reply tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: I think we have a problem w/ the baseline. When this test ran, during the RT, it would have created a debug baseline but called it ``hafs_regional_storm_following_1nest_atm_ocn``. After creating that baseline, we turned around and ran the whole rt.conf. How did the non-debug version of the test pass against that?

Comment (jiandewang) [reply tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: > Oops, you are right! Sorry, I missed that. @jiandewang would you please change this to hafs_regional_storm_following_1nest_atm_ocn_debug

fixed and committed

Comment (jkbk2004) [reply tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: I think we have to re-run hafs_regional_storm_following_1nest_atm_ocn and hafs_regional_storm_following_1nest_atm_ocn_debug to confirm.

Comment (jkbk2004) [reply tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: strange! hafs_regional_storm_following_1nest_atm_ocn_debug run has some trouble to copy over input file (grid_spec,nc).

Comment (DeniseWorthen) [reply tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: grid_spec is being copied from a directory name that includes the test name, for example:

input-data-20221101/FV3_hafs_input_data/INPUT_hafs_regional_storm_following_1nest_atm/grid_spec.nc

But we still need to understand. @jkbk2004 Did you run *only* the debug test when you tried to create the new test basline?

**Comment by junwang-noaa** [comment]
Sorry I did not create a full baseline with the new test (just create a single test baseline and run against it). I am now doing testing. Following changes are also needed to fix the issue Denise mentioned:
```
--- a/tests/fv3_conf/hafs_fv3_run.IN
+++ b/tests/fv3_conf/hafs_fv3_run.IN
@@ -16,6 +16,7 @@ if [ ${CNTL_DIR} = "hafs_regional_atm" ] || \
 elif [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm" ] || \
      [ ${CNTL_DIR} = "hafs_regional_specified_moving_1nest_atm" ] || \
      [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm_ocn" ] || \
+     [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm_ocn_debug" ] || \
      [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm_ocn_wav" ]; then
   cp -r @[INPUTDATA_ROOT]/FV3_hafs_input_data/INPUT_hafs_regional_storm_following_1nest_atm/* ./INPUT/
 elif [ ${CNTL_DIR} = "hafs_regional_1nest_atm" ] || \
diff --git a/tests/fv3_conf/hycom_hat10_run.IN b/tests/fv3_conf/hycom_hat10_run.IN
index 23000687..740e65bb 100644
--- a/tests/fv3_conf/hycom_hat10_run.IN
+++ b/tests/fv3_conf/hycom_hat10_run.IN
@@ -1,5 +1,6 @@

 if [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm_ocn" ] || \
+   [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm_ocn_debug" ] || \
    [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm_ocn_wav" ]; then
   rsync -arv @[INPUTDATA_ROOT]/FV3_hafs_input_data/HYCOM_hafs_regional_input_data/. ./
 else
```

File: tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug
Comment (jkbk2004) [reply tests/tests/hafs_regional_storm_following_1nest_atm_ocn_debug]: @jkbk2004 yes, I only ran debug.

**Comment by jkbk2004** [comment]
> [ ${CNTL_DIR} = "hafs_regional_storm_following_1nest_atm_ocn_debug" ] ||

I am running those two cases again with the changes.

**Comment by junwang-noaa** [comment]
I created hafs_regional_storm_following_1nest_atm_ocn_debug new baseline, and then run full RT against the current baseline + hafs_regional_storm_following_1nest_atm_ocn_debug baseline, the RT ran successfully.  scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20230130/hafs/ufs-weather-model/tests/RegressionTests_hera.intel.log.
The new baseline is at:
/scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20230201/INTEL

**Comment by jkbk2004** [comment]
> I created hafs_regional_storm_following_1nest_atm_ocn_debug new baseline, and then run full RT against the current baseline + hafs_regional_storm_following_1nest_atm_ocn_debug baseline, the RT ran successfully. scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20230130/hafs/ufs-weather-model/tests/RegressionTests_hera.intel.log. The new baseline is at: /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20230201/INTEL

I checked other machines. hafs_regional_storm_following_1nest_atm_ocn_debug passes ok. I am adding the code change to hafs_fv3_run.IN and hycom_hat10_run.IN.

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA baselines are identical: hafs_regional_storm_following_1nest_atm_ocn_debug and hafs_regional_storm_following_1nest_atm_ocn. On wcoss2, you can copy hafs_regional_storm_following_1nest_atm_ocn for hafs_regional_storm_following_1nest_atm_ocn_debug. But only `atm.nest02.f001.nc  atmf001.nc  sfc.nest02.f001.nc  sfcf001.nc` On Gaea, I did test hafs_regional_storm_following_1nest_atm_ocn_debug with that approach. It should pass on wcoss2 as well.

**Review by jkbk2004 — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
@jkbk2004 Can you explain how the baselines are identical? Debug should run optimizations different than non-debug which would cause different results.

**Comment by jkbk2004** [comment]
> @jkbk2004 Can you explain how the baselines are identical? Debug should run optimizations different than non-debug which would cause different results.

I think there is no rounding-off difference with -g -traceback or debug options at lower level optimization.

**Comment by jkbk2004** [comment]
I agree we should see some level of rounding-off difference by triggering different level of optimization. The difference not necessarily by debug option. I think this is another reason to double check our compile options.

**Comment by DeniseWorthen** [comment]
I still think we need to be sure about the compile, it makes no sense to me that the answers aren't different. For example, the atmf000.nc control_p8 and control_debug_p8 files are different. I think that is after a single startup timestep (ie, 000 is really after the first timestep of the atm).

**Comment by junwang-noaa** [comment]
@jkbk2004 As I explained in the slack, the results from the new HAFS debug test are **different** from the HAFS control hafs_regional_storm_following_1nest_atm_ocn results. The baseline filenames are different, and they were put in the control baseline by mistake, but the RT will still pass because the code changes have no impact on other tests and previous baseline is not touched in the full RT. Hope this clarifies the confusion.

**Comment by DeniseWorthen** [comment]
I know what is happening now. The 001 files from the debug run are still present on hera in the non-debug baseline hafs_regional_storm_following_1nest_atm_ocn. 

We should remove the files that were created by the debug test from the non-debug baseline.

**Comment by BrianCurtis-NOAA** [comment]
> I know what is happening now. The 001 files from the debug run are still present on hera in the non-debug baseline hafs_regional_storm_following_1nest_atm_ocn.
> 
> We should remove the files that were created by the debug test from the non-debug baseline.

The next PR will create new baselines, right? We can just wait for that, it won't hurt to keep the extra baselines in the non-debug directory.

**Comment by DeniseWorthen** [comment]
@BrianCurtis-NOAA Normally I might agree about the next baseline, but in this case I think the amount of confusion that has been generated makes it worthwhile to clean up.

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA Normally I might agree about the next baseline, but in this case I think the amount of confusion that has been generated makes it worthwhile to clean up.

Dogwood is cleaned up. New debug baselines are created in the correct dir. Dogwood is set for this merge.

Acorn can officially be skipped. The RT's will compile but not run, no logs are output. Seems a system issue, but it will need attention ASAP for the next set of PR's

**Comment by jkbk2004** [comment]
OK! `hafs_regional_storm_following_1nest_atm_ocn and hafs_regional_storm_following_1nest_atm_ocn_debug` are all updated ok on hera/orion/cheyenne. gaea sis down today. I will catch up gaea later. We can move to start merging process.

**Comment by BrianCurtis-NOAA** [comment]
the ecflow nodes on Dogwood went down before i could re-run one test that failed, but it passed earlier. The debug and non-debug tests passed, so WCOSS2 is good to go.

**Comment by jkbk2004** [comment]
cool! I am merging https://github.com/NOAA-EMC/fv3atm/pull/622. @jiandewang can you merging mom6 pr?

**Comment by jkbk2004** [comment]
@jiandewang https://github.com/NOAA-EMC/fv3atm/pull/622 was merged. Please, go ahead to update submodule pointers and revert the changes in gitmodules after you merge mom6 pr.

**Comment by jiandewang** [comment]
> @jiandewang [NOAA-EMC/fv3atm#622](https://github.com/NOAA-EMC/fv3atm/pull/622) was merged. Please, go ahead to update submodule pointers and revert the changes in gitmodules after you merge mom6 pr.

MOM6 merged

**Comment by jkbk2004** [comment]
@jiandewang cool! yeah, we can merge once submodule pointers (fv3/mom6) are updated and gitmodules changes are reverted.

**Comment by DeniseWorthen** [comment]
@jiandewang The hash for MOM6 needs to be [d0c4550](https://github.com/NOAA-EMC/MOM6/commit/d0c4550af8cb4dbdcebed376223a6f829a07595f)

**Review by DeniseWorthen — APPROVED** [review]

**Comment by jiandewang** [comment]
I reverted MOM6 and committed, but for FV3, this is what I did (after revert its submodule):
cd FV3
git fetch
git checkout develop

but I got many files being removing information, am I doing correct ? (I haven't committed yet)
/scratch1/NCEPDEV/climate/Jiande.Wang/MOM6-update/MOM6-update-20230206/Final/ufs-weather-model/FV3[150]git checkout develop
warning: unable to rmdir 'upp': Directory not empty
M       atmos_cubed_sphere
M       ccpp/framework
M       ccpp/physics
Previous HEAD position was ffeb762 Merge remote-tracking branch 'upstream/develop' into hafs_movingnest
Switched to branch 'develop'
Your branch and 'origin/develop' have diverged,
and have 1 and 273 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
/scratch1/NCEPDEV/climate/Jiande.Wang/MOM6-update/MOM6-update-20230206/Final/ufs-weather-model/FV3[151]git pull
Removing time_utils.F90
Removing tests/tests/fv3_stretched_nest
Removing tests/tests/fv3_stretched
Removing tests/tests/fv3_restart
Removing tests/tests/fv3_read_inc
Removing tests/tests/fv3_gfdlmp
Removing tests/tests/fv3_decomp
Removing tests/tests/fv3_control
Removing tests/tests/fv3_2threads
Removing tests/tail.h
Removing tests/run_test.sh
Removing tests/rt_utils.sh
Removing tests/rt_fv3.sh
Removing tests/rt.sh
Removing tests/rt.conf
Removing tests/head.h
Removing tests/fv3_conf/stretched-nest-input.nml.IN
Removing tests/fv3_conf/stretched-input.nml.IN
Removing tests/fv3_conf/input_nest02.nml.IN
Removing tests/fv3_conf/input.nml.IN
Removing tests/fv3_conf/gfdlmp_run.IN
Removing tests/fv3_conf/gfdlmp.nml.IN
Removing tests/fv3_conf/fv3_stretched_run.IN
Removing tests/fv3_conf/fv3_run.IN
Removing tests/fv3_conf/fv3_qsub.IN_theia
Removing tests/fv3_conf/fv3_bsub.IN_wcoss_cray
Removing tests/fv3_conf/fv3_bsub.IN
Removing tests/detect_machine.sh
Removing tests/default_vars.sh
Removing tests/compile.sh
Removing tests/atparse.bash
Removing tests/abort_dep_tasks.py
Removing tests/RegressionTests_wcoss_cray.log
Removing tests/RegressionTests_theia.log
Removing tests/Compile_wcoss_cray.log
Removing tests/Compile_theia.log
Removing namphysics/physics/shalcnv.f
Removing namphysics/physics/sascnvn.f
Removing namphysics/physics/radsw_param.f
Removing namphysics/physics/radsw_main_nmmb.f
Removing namphysics/physics/radsw_datatb.f
Removing namphysics/physics/radlw_param.f
Removing namphysics/physics/radlw_main_nmmb.f
Removing namphysics/physics/radlw_datatb.f
Removing namphysics/physics/radiation_surface_nmmb.f
Removing namphysics/physics/radiation_gases_nmmb.f
Removing namphysics/physics/radiation_clouds_nmmb.f
Removing namphysics/physics/radiation_astronomy_nmmb.f
Removing namphysics/physics/radiation_aerosols_nmmb.f
Removing namphysics/physics/rad_initialize_nmmb.f
Removing namphysics/physics/precpd_nmmb.f
Removing namphysics/physics/physparam.f
Removing namphysics/physics/physcons.f90
Removing namphysics/physics/ozne_def.f
Removing namphysics/physics/noahmp_tables.f90
Removing namphysics/physics/namelist_soilveg.f
Removing namphysics/physics/moninq.f
Removing namphysics/physics/moninedmf.f
Removing namphysics/physics/module_mp_thompson.F90
Removing namphysics/physics/module_mp_radar.F90
Removing namphysics/physics/module_TURBULENCE.F90
Removing namphysics/physics/module_SF_URBAN.F90
Removing namphysics/physics/module_SF_JSFC.F90
Removing namphysics/physics/module_SF_GFDL.F90
Removing namphysics/physics/module_RA_RRTM.F90
Removing namphysics/physics/module_RA_GFDL.F90
Removing namphysics/physics/module_RADIATION.F90
Removing namphysics/physics/module_MP_WSM6.F90
Removing namphysics/physics/module_MP_GFS.F90
Removing namphysics/physics/module_MP_FER_HIRES.F90
Removing namphysics/physics/module_MP_ETANEW.F90
Removing namphysics/physics/module_MICROPHYSICS.F90
Removing namphysics/physics/module_LS_NOAHLSM.F90
Removing namphysics/physics/module_LS_LISS.F90
Removing namphysics/physics/module_KINDS.F90
Removing namphysics/physics/module_GWD.F90
Removing namphysics/physics/module_DIAGNOSE.F90
Removing namphysics/physics/module_CU_SCALE.F90
Removing namphysics/physics/module_CU_SASHUR.F90
Removing namphysics/physics/module_CU_SAS.F90
Removing namphysics/physics/module_CU_BMJ.F90
Removing namphysics/physics/module_CONVECTION.F90
Removing namphysics/physics/module_CONTROL.F90
Removing namphysics/physics/module_CONSTANTS.F90
Removing namphysics/physics/module_BL_MYJPBL.F90
Removing namphysics/physics/module_BL_GFSPBLHUR.F90
Removing namphysics/physics/module_BL_GFSPBLEDMFHUR.F90
Removing namphysics/physics/module_BL_GFSPBL.F90
Removing namphysics/physics/mfshalcnv.f
Removing namphysics/physics/mfpbl.f
Removing namphysics/physics/mfdeepcnv.f
Removing namphysics/physics/mersenne_twister.f
Removing namphysics/physics/machine.F
Removing namphysics/physics/iounitdef.f
Removing namphysics/physics/h2o_def.f
Removing namphysics/physics/gscond.f
Removing namphysics/physics/grrad_nmmb.f
Removing namphysics/physics/gfdl_cloud_microphys.F90
Removing namphysics/physics/funcphys.f90
Removing namphysics/physics/GFDL_parse_tracers.F90
Removing namphysics/makefile
Removing namphysics/NAM_layer/module_SOLVER_GRID_COMP.F90
Removing namphysics/NAM_layer/NAM_typedefs.F90
Removing namphysics/NAM_layer/NAM_restart.F90
Removing namphysics/NAM_layer/NAM_radiation_driver.F90
Removing namphysics/NAM_layer/NAM_driver.F90
Removing namphysics/NAM_layer/NAM_diagnostics.F90
Removing namphysics/NAM_layer/NAM_abstraction_layer.F90
Removing mkDepends.pl
Removing makefile
Removing ipd/makefile
Removing ipd/IPD_typedefs.F90
Removing ipd/IPD_driver.F90
Removing io/post_gfs_stub.F90
Removing io/post_gfs.F90
Removing io/module_write_nemsio.F90
Removing io/makefile
Removing io/ffsync.F90
Removing gfsphysics/physics/wv_saturation.F
Removing gfsphysics/physics/wam_f107_kp_mod.f90
Removing gfsphysics/physics/ugwp_driver_v0.f
Removing gfsphysics/physics/tridi2t3.f
Removing gfsphysics/physics/tracer_const_h.f
Removing gfsphysics/physics/surface_perturbation.F90
Removing gfsphysics/physics/shalcv_opr.f
Removing gfsphysics/physics/shalcv_fixdp.f
Removing gfsphysics/physics/shalcv_1lyr.f
Removing gfsphysics/physics/shalcv.f
Removing gfsphysics/physics/shalcnv.f
Removing gfsphysics/physics/sflx.f
Removing gfsphysics/physics/sfcsub.F
Removing gfsphysics/physics/sfc_sice.f
Removing gfsphysics/physics/sfc_ocean.f
Removing gfsphysics/physics/sfc_nst.f
Removing gfsphysics/physics/sfc_noahmp_drv.f
Removing gfsphysics/physics/sfc_drv.f
Removing gfsphysics/physics/sfc_diff.f
Removing gfsphysics/physics/sfc_diag.f
Removing gfsphysics/physics/sfc_cice.f
Removing gfsphysics/physics/set_soilveg.f
Removing gfsphysics/physics/satmedmfvdifq.f
Removing gfsphysics/physics/satmedmfvdif.f
Removing gfsphysics/physics/sascnvn.f
Removing gfsphysics/physics/sascnv.f
Removing gfsphysics/physics/samfshalcnv.f
Removing gfsphysics/physics/samfdeepcnv.f
Removing gfsphysics/physics/samfaerosols.f
Removing gfsphysics/physics/rayleigh_damp_mesopause.f
Removing gfsphysics/physics/rayleigh_damp.f
Removing gfsphysics/physics/rascnvv2.f
Removing gfsphysics/physics/radsw_param.f
Removing gfsphysics/physics/radsw_main.f
Removing gfsphysics/physics/radsw_datatb.f
Removing gfsphysics/physics/radlw_param.f

**Comment by BrianCurtis-NOAA** [comment]
@jiandewang I'm going to guess you checked out your own develop branch and not the one from FV3? you need to add https://github.com/NOAA-EMC/fv3atm to your remotes, update it, then `git checkout <NOAA-EMC/fv3atm remote name> develop`

i.e. in FV3:
```
git remote add upstream https://github.com/NOAA-EMC/fv3atm
git remote update
git checkout upstream develop (or maybe upstream/develop)
cd ..
git add FV3
git commit -m "Update FV3 hash"
git push origin update-MOM6-20230206
```

**Comment by junwang-noaa** [comment]
After: git checkout upstream develop (or maybe upstream/develop), please
also add
git submodule update --init --recursive
to get all the subcomponents.

**Comment by jiandewang** [comment]
many thanks to Denise's help, FV3 and MOM6 are pointing to the right hash # now

**Pull request merged**
